libs/clients/auth-public-api/src/lib/auth-public-api-client.module.ts

```diff
@@ -1,37 +1,58 @@
 import { DynamicModule } from '@nestjs/common'
-import { Configuration, DelegationsApi, ApiScopeApi } from '../../gen/fetch'
+
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+import {
+  Configuration,
+  MeDelegationsApi,
+  ScopesApi,
+  ActorDelegationsApi,
+} from '../../gen/fetch'
 
 export interface AuthPublicApiClientModuleConfig {
   baseApiUrl: string
 }
 
 export class AuthPublicApiClientModule {
   static register(config: AuthPublicApiClientModuleConfig): DynamicModule {
+    const fetch = createEnhancedFetch({
+      name: 'auth-public-api-client',
+    })
+
     return {
       module: AuthPublicApiClientModule,
       providers: [
         {
-          provide: DelegationsApi,
+          provide: MeDelegationsApi,
+          useFactory: () =>
+            new MeDelegationsApi(
+              new Configuration({
+                fetchApi: fetch,
+                basePath: config.baseApiUrl,
+              }),
+            ),
+        },
```

Comment:
Fyi. In the national registry client we do this as a `...apis.map()` expression, which reduces repitition.

Reply:
Nice, did that and moved the config to use the new ConfigModule